src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -676,6 +677,32 @@ public final SAMFileGATKReadWriter createSAMWriter(final File outputFile, final
         );
     }
 
+    /*
+     * Create a common SAMFileWriter using the reference and read header for this tool.
+     *
+     * @param outputFile    - if this file has a .cram extension then a reference is required. Can not be null.
+     * @param preSorted     - if true then the records must already be sorted to match the header sort order
+     *
+     * @throws UserException if outputFile ends with ".cram" and no reference is provided
+     * @return SAMFileWriter
+     */
+    public final SAMFileGATKReadWriter createSAMWriter(final Path outputPath, final boolean preSorted) {
```

Comment:
I think the old file methods should delegate to these new ones.   Wrap the file in `Utils.nonNull(file).toPath()` to avoid introducing NPEs where we used to have `IllegalArgumentException`.

Reply:
The two methods are not exactly identical, sadly, so if we do this then we're changing the semantics of the old ones.